Note fading and persistent messages

MrsTonedOne opened the pull request:
Player Notes can now fade out over time so admins don't accidentally ban players for 4 year old notes (admins may still reveal these older notes)

Deleted messages now remain in the database and are instead just hidden from view.

Two things relating to messages I've been meaning to do for ever.
:cl:
add: Player notes can now be configured to fade out over time to allow admins to quickly see how recent notes are. Server Operators, check the config for more info
/:cl:

30 to 90 day fade out preview:

![1](https://user-images.githubusercontent.com/7069733/29907449-3b2fed56-8dd0-11e7-9c11-9bf2e67e0645.PNG)
![2](https://user-images.githubusercontent.com/7069733/29907448-3b2f11ce-8dd0-11e7-8ff9-5ab77d7875a5.PNG)


Defaults to disabled on existing servers, defaults to a fade out between 3 months to a year on new servers (What I plan on using for /tg/)

MrsTonedOne committed:
Note fading and persistent messages
Player Notes can now fade out over time so admins don't accidentally ban players for 4 year old notes
Deleted messages now remain in the database and are instead just hidden from view.

Two things relating to messages I've been meaning to do for ever.

MrStonedOne commented:
This will conflict when the other sql prs are merged. I'll just update the version and fix the changelog conflicts when that happens

Jordie0608 commented on SQL/database_changelog.txt:
```diff
@@ -1,10 +1,25 @@
+Modified table 'messages', adding a deleted column and editing all indexes to include it
+
+ALTER TABLE `messages`
+ADD COLUMN `deleted` tinyint(1) NOT NULL DEFAULT '0' AFTER `edits`,
```
You missed `unsigned` here.

Jordie0608 commented on code/controllers/configuration.dm:
```diff
@@ -102,6 +102,8 @@
 	var/use_age_restriction_for_jobs = 0 //Do jobs use account age restrictions? --requires database
 	var/use_account_age_for_jobs = 0	//Uses the time they made the account for the job restriction stuff. New player joining alerts should be unaffected.
 	var/see_own_notes = 0 //Can players see their own admin notes (read-only)? Config option in config.txt
+	var/note_fresh_days
```
Comments here ploxkthxbye

lzimann replied:
probably needs a default value aswell

MrStonedOne replied:
nope, the default is set to null intentionally. 0 = start fading out immediately, null = disabled.

Jordie0608 reviewed (changes requested):
I feel like someone's going to miss hidden notes if they fade entirely; aside from not making them disappear once old enough, what about a counter next to the notes header of total/hidden

i.e.
`Notes (34, 18 Hidden)`
It might get confused with secret/non-secret however, i' not sure what might be better wording.

MrStonedOne commented:
Jordie, could be some text next to the show all button

Jordie0608 commented:
Obligatory
![](http://i0.kym-cdn.com/photos/images/facebook/000/565/679/781.png)

kevinz000 commented:
>admins not knowing how to read dates

oh come on

lzimann commented on config/config.txt:
```diff
@@ -241,6 +241,12 @@ TICKLAG 0.5
 
+### Comment these two out to prevent notes fading out over time for admins.
+## Notes older then this will start fading out.
+NOTE_FRESH_DAYS 91.31055
```
such a strange number on these

MrStonedOne replied:
that is 3 avg months.

lzimann replied:
I meant the `.31055` part(same for the next number)

RemieRichards replied:
as he said, it's an average.
a month is often taken to be 4 weeks but this is not consistent, some months have 28 days, others have 31.

MrStonedOne replied:
1 year = 365.2422 days (to account for leap years)

1 month = 365.2422/12 = 30.43685 days

30.43685 * 3 = 91.31055

Cyberboss commented:
You know it's true kev

MrsTonedOne committed:
Make the fact there are hidden notes more obvious.
We print the first note that would be skipped then pixel shift the show all button over the skipped noted.

The rest after that are not printed

MrStonedOne commented:
I came up with a better solution.

MrsTonedOne committed:
Actually, lets make it say how many are hidden.

Cyberboss commented on SQL/tgstation_schema_prefixed.sql:
```diff
@@ -31,7 +31,7 @@ CREATE TABLE `SS13_admin` (
 --
 
-DROP TABLE IF EXISTS `SS13_dmin_log`;
+DROP TABLE IF EXISTS `SS13_admin_log`;
```
HA

Jordie0608 replied:
Maybe we should leave it just so nobody tries to use this system.
